### PR TITLE
Feat/voice active

### PR DESCRIPTION
**Pull request by esanchez411**

### CODE REVIEW

File: src/pages/agentInteraction/agentInteraction.tsx [hunk 1]
```diff
@@ -47,23 +47,17 @@ const AgentInteraction: React.FC<AgentInteractionProps> = ({ bot, onBack }) => {
...
   }, [bot.persona_id]);
 
-  useEffect(() => {
-    const fetchMessages = async () => {
-      if (conversationId) {
+    const fetchMessages = async (itemId: string) => {
         try {
           const messagesServiceInstance = new messagesService();
-          const response = await messagesServiceInstance.getConversationMessages(conversationId);
+          const response = await messagesServiceInstance.getConversationMessages(itemId);
```
Comment (rquesada) [inline src/pages/agentInteraction/agentInteraction.tsx:53]: Why itemId and not conversationId?

File: src/pages/home/botsGrid/botsGrid.tsx [hunk 2]
```diff
@@ -61,7 +61,16 @@ const BotsGrid: React.FC<BotsGridProps> = ({ viewType, handleAddNewAgent, onBotS
   return (
     <>
       <ToastContainer />
-      <BotListTitle viewType={viewType} onAddNewAgent={handleAddNewAgent} />
+      <BotListTitle 
+        viewType={viewType} 
+        onAddNewAgent={() => {
+          if ((viewType === AgentViewType.MyAgents || viewType === AgentViewType.SharedAgents) && personas.length === 0) {
```
Comment (rquesada) [inline src/pages/home/botsGrid/botsGrid.tsx:67]: Extract the logic of "if" to a method in order to be more readable:

Create a method `canCreateAgent` like:
```
const canCreateAgent = () => {
  return (viewType === AgentViewType.MyAgents || viewType === AgentViewType.SharedAgents) && personas.length === 0;
};
```

Replace like:
```
const handleAddAgentClick = () => {
  if (canCreateAgent()) {
    toast.error("Cannot create an assistant: please create a persona first");
  } else {
    handleAddNewAgent();
  }
};
```

File: src/data/types.ts [hunk 3]
```diff
@@ -49,6 +49,7 @@ export interface Message {
   timestamp: string;
   prompt: string;
   chat_response: string;
+  voice_active: string;
```
Comment (rquesada) [inline src/data/types.ts:52]: nit: rename with `is_voice_active` in order to know that it is boolean value

**Review by rquesada — CHANGES_REQUESTED** [review]

File: src/pages/home/botsGrid/botsGrid.tsx (hunk 2 above)
Comment (esanchez411) [reply src/pages/home/botsGrid/botsGrid.tsx:67]: Done

File: src/data/types.ts (hunk 3 above)
Comment (esanchez411) [reply src/data/types.ts:52]: Done

File: src/pages/agentInteraction/agentInteraction.tsx (hunk 1 above)
Comment (esanchez411) [reply src/pages/agentInteraction/agentInteraction.tsx:53]: Done

**Review by rquesada — APPROVED** [review]
lgtm

**Pull request merged**